client/lib/widgets/course_tile.dart

```diff
@@ -41,9 +46,27 @@ class CourseTile extends StatelessWidget {
                           )),
                     ),
                   ),
-                  Container(
-                      margin: const EdgeInsets.only(top: 7, right: 6),
-                      child: const Icon(Icons.star_border, size: 20)),
+                  InkWell(
+                    onTap: () async {
+                      if (course['roles'][uid]['favorite'] == null) {
+                        await addCourseToFavorite(course.id, uid);
+                      } else {
+                        if (course['roles'][uid]['favorite']) {
+                          await removeCourseFromFavorite(course.id, uid);
+                        } else {
+                          await addCourseToFavorite(course.id, uid);
+                        }
+                      }
+                    },
+                    child: Container(
+                        margin: const EdgeInsets.only(top: 7, right: 6),
+                        child: course['roles'][uid]['favorite'] == null
+                            ? const Icon(Icons.star_border, size: 20)

```

Comment:
can maybe simplify this double ternary to something like 
```
course['roles'][uid]['favorite'] == null || course['roles'][uid]['favorite'] == false ? 
const Icon(Icons.star_border, size: 20) : const Icon(Icons.star, color: Colors.amber, size: 20)
```

def could probably be made cleaner than even that but just for now

Reply:
makes sense, wasn't sure if || short circuits in dart